CMakeLists.txt

```diff
@@ -257,6 +257,40 @@ configure_file(programs/cleos/LICENSE.CLI11                                licen
 
 install(DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/licenses/leap" DESTINATION "${CMAKE_INSTALL_FULL_DATAROOTDIR}/licenses/" COMPONENT base)
 
+install(DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/libraries/testing/contracts DESTINATION ${CMAKE_INSTALL_FULL_DATAROOTDIR}/leap_testing/libraries/testing COMPONENT dev
```

Comment:
Unfortunately, for any place you have a `COMPONENT dev` it should also be `EXCLUDE_FROM_ALL`.

The problem is, we want `make install` to only install the files from the base component (or put another way, we want `make install` to ignore anything we classify as a dev install), but there is no way afaik to control the behavior of `make install` since it's a built in target that can't be overridden. So we have to resort to peeling away all these from the "install all" target.

The reason we go through this trouble is that historically, before the `COMPONENT dev EXCLUDE_FROM_ALL` behavior, a `make install` could install header files that were in a path with higher precedence than our own source tree. As you can imagine this led to users (and devs!) getting in to really wonky problems when the, for example, fc headers in /usr/local/include would be used instead of the fc headers in the current source tree. There is a little more about discussion on this in eosnetworkfoundation/mandel#277

Reply:
Addressed: [be154da](https://github.com/AntelopeIO/leap/pull/818/commits/be154da6036dbe810631b53b32afee58491767fc)